modyn/supervisor/internal/eval/strategies/abstract.py

```diff
@@ -1,9 +1,25 @@
 from abc import ABC, abstractmethod
-from typing import Iterable, Optional
+from dataclasses import dataclass
+from typing import Iterable
 
 from modyn.config.schema.pipeline import EvalStrategyConfig
 
 
+@dataclass
+class EvalInterval:
+    start: int | None
+    """The start timestamp of the evaluation interval handed over to the evaluation server."""
+
+    end: int | None
+    """The end timestamp of the evaluation interval handed over to the evaluation server."""
+
+    training_interval_start: int
+    """The first timestamp in the training request."""
+
+    training_interval_end: int
+    """The last timestamp in the training request."""
```

Comment:
I do not understand this yet. Why is this part of eval interval? Why can this not be None, but start and end can? From my understanding, any eval interval was some slice of a dataset, but now we have 4 parameters

Reply:
yeah, this is weird, see my comment below. We might want to discuss this in person.